staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -188,8 +188,15 @@ func NewOnAddresses(dialer httpstream.Dialer, addresses []string, ports []string
 func (pf *PortForwarder) ForwardPorts() error {
 	defer pf.Close()
 
-	var err error
-	pf.streamConn, _, err = pf.dialer.Dial(PortForwardProtocolV1Name)
+	var (
+		remotePorts []uint16
+		err         error
+	)
+	for _, port := range pf.ports {
+		remotePorts = append(remotePorts, port.Remote)
+	}
+
+	pf.streamConn, _, err = pf.dialer.Dial(remotePorts, PortForwardProtocolV1Name)
```

Comment:
adding the data at this layer is depending on the client to self-report the ports it is dialing in headers... that's not reliable from an audit perspective (the client could report port 8080 here, then establish a connection to a different port)